### PR TITLE
[Evil] Implementing dirname

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
This PR aims to implement the missing `dirname` in some way that it should work when using multiple threads.

### CODE REVIEW

File: src/lib/evil/evil_dirname.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+#include <evil_api.h>
```
Comment (walac) [inline src/lib/evil/evil_dirname.c:5]: nit: this was already included in `evil_dirname.h`

File: src/lib/evil/evil_dirname.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <stdlib.h>
+
+#include <evil_windows.h>
+
+EVIL_API char* dirname (char* path)
+{
+  static char dir[MAX_PATH];
```
Comment (walac) [inline src/lib/evil/evil_dirname.c:13]: Use a threa local storage, otherwise this can't be called from multiple threads.

Comment (joaoantoniocardoso) [reply src/lib/evil/evil_dirname.c:13]: Oh then we should fix evil's `basename()` too.

File: src/lib/evil/evil_libgen.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#ifndef __EVIL_LIBGEN_H__
+#define __EVIL_LIBGEN_H__
+
+#include <evil_private.h>
+
+#ifdef _MSC_VER
+# define dirname evil_dirname
+#endif
```
Comment (Coquinho) [inline src/lib/evil/evil_libgen.h:8]: I bring the same question that I brought at `evil_basename`, should we use [_MSC_VER or HAVE_FOO](https://github.com/expertisesolutions/efl/pull/323#discussion_r458078126) and why?

**Pull request merged**